plugins/ArchivingMetrics/tests/Integration/TimerDbTest.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link    https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Plugins\ArchivingMetrics\tests\Integration;
+
+use Piwik\Common;
+use Piwik\Date;
+use Piwik\Db;
+use Piwik\Period;
+use Piwik\Plugins\ArchivingMetrics\Clock\Clock;
+use Piwik\Plugins\ArchivingMetrics\Context;
+use Piwik\Plugins\ArchivingMetrics\Timer;
+use Piwik\Plugins\ArchivingMetrics\Writer\DbWriter;
+use Piwik\Segment;
+use Piwik\Tests\Framework\TestCase\IntegrationTestCase;
+
+/**
+ * @group ArchivingMetrics
+ * @group ArchivingMetrics_TimerDb
+ * @group Plugins
+ */
+class TimerDbTest extends IntegrationTestCase
+{
+    public function setUp(): void
+    {
+        Db::query('DELETE FROM ' . Common::prefixTable('archiving_metrics'));
+    }
+
+    public function testItWritesAndReadsFromDatabase(): void
+    {
+        $period = new Period\Day(Date::factory('2025-11-01'));
+        $segment = $this->createMock(Segment::class);
+
+        $context = new Context(1, $period, $segment, '');
+
+        $timer = new Timer(true, new Clock(), new DbWriter());
+        $timer->start($context);
+        $timer->complete($context, [999], false);
+
+        $rows = Db::fetchAll('SELECT * FROM ' . Common::prefixTable('archiving_metrics'));
+
+        $this->assertNotEmpty($rows, 'Expected archiving_metrics table to have at least one record.');
```

Comment:
```suggestion
        self::assertCount(1, $rows, 'Expected archiving_metrics table to have exactly one record.');
```

The database should be reset by the test suite, and we are calling the function exactly once. Having more than one row in the database could point at something going wrong.